### PR TITLE
Nn 2342 throw error if attendance exists

### PR DESCRIPTION
**Pull request by Nimphal**
NN-2342: Throw error if user tries to create an attendance when one already exists

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/whereabouts/controllers/AttendanceController.kt [hunk 1]
```diff
@@ -30,9 +31,14 @@ class AttendanceController(private val attendanceService: AttendanceService) {
       @ApiParam(value = "Attendance details", required = true)
       @RequestBody
       @Valid
-      attendance: CreateAttendanceDto): AttendanceDto {
+      attendance: CreateAttendanceDto): ResponseEntity<Any> {
+    val createdAttendance: AttendanceDto =  try {
+      attendanceService.createAttendance(attendance)
+    } catch (e: AttendanceExists) {
+      return ResponseEntity.badRequest().build()
```
Comment (JamieDraperUK) [inline src/main/java/uk/gov/justice/digital/hmpps/whereabouts/controllers/AttendanceController.kt]: Should we return a http 409?

Comment (Nimphal) [reply src/main/java/uk/gov/justice/digital/hmpps/whereabouts/controllers/AttendanceController.kt]: We definitely should, good catch!

Comment (JamieDraperUK) [reply src/main/java/uk/gov/justice/digital/hmpps/whereabouts/controllers/AttendanceController.kt]: Might be worth returning an error message too.

Comment (Nimphal) [reply src/main/java/uk/gov/justice/digital/hmpps/whereabouts/controllers/AttendanceController.kt]: 👍

**Pull request merged**